### ISSUE TITLE
Re-installation errors

### ISSUE THREAD

**Issue by focussing** [SimulatedGREG/electron-vue#255]
Suddenly my electron-vue environment was giving errors at npm run dev after having updated via npm update.
Deleted the folder
Updated vue-init to latest version 2.8.2
Reinstalled the boilerplate -> Errors as below
Nodejs version is not changed and is still at 4.6.2
NPM version is not changed, still at 4.5.0
Please help...

```
macbook:sites raymond$ vue init simulatedgreg/electron-vue electron-vue


/Users/raymond/.vue-templates/simulatedgreg-electron-vue/meta.js:3
const { join } = require('path')
      ^

SyntaxError: Unexpected token {
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at getMetadata (/Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/lib/options.js:43:15)
    at options (/Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/lib/options.js:15:14)
    at generate (/Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/lib/generate.js:36:14)
    at /Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/bin/vue-init:153:5
    at /Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/node_modules/download-git-repo/index.js:43:22
    at /Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/node_modules/download/index.js:165:3
    at callback (/Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/node_modules/each-async/index.js:38:4)
    at onetime (/Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/node_modules/onetime/index.js:22:12)
    at /Users/raymond/.nvm/versions/node/v4.6.2/lib/node_modules/vue-cli/node_modules/download/index.js:156:5


```

**Comment by SimulatedGREG** [comment]
electron-vue requires `node@^7` which support destructuring.

**Comment by focussing** [comment]
since when? it was working flawlessly

**Comment by SimulatedGREG** [comment]
Since not too long ago. This boilerplate is always constantly introducing features. And with the current setup available in the `master` branch you will need `node@^7`. 

If you are in need of having multiple installions of node for you system I'd recommend using http://nvm.sh. If you want to see the new features available make sure to check out the docs and #171 for info about the latest milestone release.

**Comment by focussing** [comment]
That makes sense off course, however such breaking updates are not nice...
So I ruined my setup when npm updating?
I would be switching between node 4.6.2 and 7 all day... My main development tool requires 4.6.2 max (Appcelerator Titanium).
How can I install the boilerplate that was previously working with node 4.6.2?
Or would I be missing a lot then?

**Comment by SimulatedGREG** [comment]
> So I ruined my setup when npm updating?

By running `npm update`, yes there is a chance that could break the setup. Because the versions defined in `package.json` are caret ranges there is always a slim possibility that a newer version of specific a module(s) could break compatibility with others. This is why package managers like `yarn` and `npm@5` introduced a lock file architecture to "lock-in" specific versions of modules for your local installations to be reliable.

> I would be switching between node 4.6.2 and 7 all day... My main development tool requires 4.6.2 max (Appcelerator Titanium).

I would highly suggest to make a way to upgrade your node version. Not just because electron-vue recommends it, but because you also get more security patches and features that are always important to have. I'm sure you're very aware of this, but even LTS is at 6.11.0. 

> How can I install the boilerplate that was previously working with node 4.6.2?

There isn't an easy way to accomplish mainly because `vue-cli` wasn't built to scaffold from a specific commit. But what I would suggest is if you check this previously [`package.json`](https://github.com/SimulatedGREG/electron-vue/blob/633f6464f1ca346decbde56a035fee14a3dfeebd/template/package.json)(taken from a `package.json` previous to Milestone Minimize) and remove all `^` definitions and try installing with that. I can't guarantee that will work, but its a good direction to head towards.

> Or would I be missing a lot then?

In short, yes. electron-vue at its current state provides much more support for the `main` process, re-written development and build scripts, automated CI deployments, target ES6 features to remove the need of transpiling completely down to ES5, and so much more. Here's a general overview of that past milestone (https://simulatedgreg.gitbooks.io/electron-vue/content/en/new-releases.html).

**Comment by focussing** [comment]
Greg, thanks a lot for your comprehensive answer. After reading all this I installed node 7 en use that via nvm. I am already rebuilding my application and reading your gitbook. This is very interesting and promising. One last suggestion; catch the error when the installed node version too low. That would helped me a lot.
